Path: bbn.com!bbn!husc6!mit-eddie!ll-xn!ames!elroy!devvax!lroot
From: lroot@devvax.JPL.NASA.GOV (The Superuser)
Newsgroups: comp.sources.bugs
Subject: perl 1.0 patch #4
Summary: This is an official patch for perl 1.0.  Please apply it.
Message-ID: <1152@devvax.JPL.NASA.GOV>
Date: 26 Jan 88 16:54:21 GMT
Organization: Jet Propulsion Laboratory, Pasadena, CA
Lines: 90

System: perl version 1.0
Patch #: 4
Priority: LOW to MEDIUM, depending
Subject: make depend doesn't work if . isn't in your PATH
From: eggert@sea.sm.unisys.com (Paul Eggert)

Description:
	make depend doesn't work if . isn't in your PATH.

Fix:	From rn, say "| patch -d DIR", where DIR is your perl source
	directory.  Outside of rn, say "cd DIR; patch <thisarticle".
	If you don't have the patch program, apply the following by hand,
	or get patch.

	If patch indicates that patchlevel is the wrong version, you may need
	to apply one or more previous patches, or the patch may already
	have been applied.  See the patchlevel.h file to find out what has or
	has not been applied.  In any event, don't continue with the patch.

	If you are missing previous patches they can be obtained from me:

	Larry Wall
	lwall@jpl-devvax.jpl.nasa.gov

	If you send a mail message of the following form it will greatly speed
	processing:

	Subject: Command
	@SH mailpatch PATH perl 1.0 LIST
		   ^ note the c

	where PATH is a return path FROM ME TO YOU in Internet notation, and
	LIST is the number of one or more patches you need, separated by spaces,
	commas, and/or hyphens.  Saying 35- says everything from 35 to the end.

	You can also get the patches via anonymous FTP from
	jpl-devvax.jpl.nasa.gov (128.149.8.43).

Index: patchlevel.h
Prereq: 3
1c1
< #define PATCHLEVEL 3
---
> #define PATCHLEVEL 4
 
Index: Makefile.SH
Prereq: 1.0.1.1
*** Makefile.SH.old	Tue Jan 26 08:48:31 1988
--- Makefile.SH	Tue Jan 26 08:48:32 1988
***************
*** 14,22 ****
  esac
  echo "Extracting Makefile (with variable substitutions)"
  cat >Makefile <<!GROK!THIS!
! # $Header: Makefile.SH,v 1.0.1.1 88/01/24 03:55:18 root Exp $
  #
  # $Log:	Makefile.SH,v $
  # Revision 1.0.1.1  88/01/24  03:55:18  root
  # patch 2: remove extra Log lines.
  # 
--- 14,25 ----
  esac
  echo "Extracting Makefile (with variable substitutions)"
  cat >Makefile <<!GROK!THIS!
! # $Header: Makefile.SH,v 1.0.1.2 88/01/26 08:46:04 root Exp $
  #
  # $Log:	Makefile.SH,v $
+ # Revision 1.0.1.2  88/01/26  08:46:04  root
+ # patch 4: make depend didn't work right if . wasn't in PATH.
+ # 
  # Revision 1.0.1.1  88/01/24  03:55:18  root
  # patch 2: remove extra Log lines.
  # 
***************
*** 137,143 ****
  	lint $(lintflags) $(defs) $(c) > perl.fuzz
  
  depend: makedepend
! 	makedepend
  
  test: perl
  	chmod 755 t/TEST t/base.* t/comp.* t/cmd.* t/io.* t/op.*
--- 140,146 ----
  	lint $(lintflags) $(defs) $(c) > perl.fuzz
  
  depend: makedepend
! 	./makedepend
  
  test: perl
  	chmod 755 t/TEST t/base.* t/comp.* t/cmd.* t/io.* t/op.*
